tools/clang/lib/SPIRV/EmitVisitor.cpp

```diff
@@ -1343,13 +1343,16 @@ bool EmitVisitor::visit(SpirvDebugTypeArray *inst) {
   curInst.push_back(inst->getDebugOpcode());
   curInst.push_back(
       getOrAssignResultId<SpirvInstruction>(inst->getElementType()));
-  for (auto it = inst->getElementCount().rbegin();
-       it != inst->getElementCount().rend(); ++it) {
+
+  // This is a reverse order of dimensions, thereby emitting in a reverse order.
+  llvm::SmallVector<uint32_t, 2> counts = inst->getElementCount();
```

Comment:
llvm::SmallVector has rbegin and rend methods. Using those makes the code much easier to read. (go back to what it was doing before).

Reply:
done